### PR TITLE
Added program for BST operations, Set-1 (#49)

### PR DESCRIPTION
**Pull request by dhairyakhale**
Covered operations insertion, search, updation, deletion (Issue #49 )

### CODE REVIEW

**Comment by dhairyakhale** [comment]
Please let me know if I made some mistake. @sahilbansal17

File: Data Structures/Trees/BST_insert-traverse-search-delete.cpp [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    // The tree is like:
+    //     5
+    //     /\
+    //    4  9
+    //   /   /\
+    //  1   7  11
+    //   \    
+    //    3
+
+    root =  bst.insertion(5, root);    //First insertion becomes root
```
Comment (aditya81070) [inline Data Structures/Trees/BST_insert-traverse-search-delete.cpp]: Take input from the user and also give a proper prompt for this.

Comment (dhairyakhale) [reply Data Structures/Trees/BST_insert-traverse-search-delete.cpp]: Should I? Then I'll have to remove the example tree also.

Comment (aditya81070) [reply Data Structures/Trees/BST_insert-traverse-search-delete.cpp]: @dhairyakhale This is also a good way.

File: Data Structures/Trees/BST_insert-traverse-search-delete.cpp [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+    return 0;
+}
+
+//Function Definitions:
+
+//-------------------INSERTION--------------------
+
+//O(n) worst case, when a BST is in form of a linked list.
+//Average complexity is O(log n), as it keeps splitting the tree in left/right sub-trees
```
Comment (ramitsawhney27) [inline Data Structures/Trees/BST_insert-traverse-search-delete.cpp]: Recurrence instead of "splitting the tree.."

Comment (dhairyakhale) [reply Data Structures/Trees/BST_insert-traverse-search-delete.cpp]: Added the requested changes. Please review.

**Comment by dhairyakhale** [comment]
Please review changes @ramitsawhney27 @sahilbansal17 @arnav-mandal1234

**Comment by ramitsawhney27** [comment]
I'm not sure where you've added the recurrence relations that show how you arrived at O(logN).

**Comment by dhairyakhale** [comment]
@ramitsawhney27 Do I write recurrence relations for all the algorithms/functions used?

**Comment by dhairyakhale** [comment]
@sahilbansal17 It is failing check because of an empty line, which does not exist in the file I committed.

**Pull request merged**